File: C#/API/Model/RunMode.cs
// ----------------------------------------------------------------------------------------------------------
// Copyright 2018 - RoboDK Inc. - https://robodk.com/
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
// http://www.apache.org/licenses/LICENSE-2.0
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
// ----------------------------------------------------------------------------------------------------------

// ----------------------------------------------------------------------------------------------------------
// This file (RoboDK.cs) implements the RoboDK API for C#
// This file defines the following classes:
//     Mat: Matrix class, useful pose operations
//     RoboDK: API to interact with RoboDK
//     RoboDK.Item: Any item in the RoboDK station tree
//
// These classes are the objects used to interact with RoboDK and create macros.
// An item is an object in the RoboDK tree (it can be either a robot, an object, a tool, a frame, a program, ...).
// Items can be retrieved from the RoboDK station using the RoboDK() object (such as RoboDK.GetItem() method) 
//
// In this document: pose = transformation matrix = homogeneous matrix = 4x4 matrix
//
// More information about the RoboDK API for Python here:
//     https://robodk.com/doc/en/RoboDK-API.html
//     https://robodk.com/doc/en/PythonAPI/index.html
//
// More information about RoboDK post processors here:
//     https://robodk.com/help#PostProcessor
//
// Visit the Matrix and Quaternions FAQ for more information about pose/homogeneous transformations
//     http://www.j3d.org/matrix_faq/matrfaq_latest.html
//
// This library includes the mathematics to operate with homogeneous matrices for robotics.
// ----------------------------------------------------------------------------------------------------------



namespace RoboDk.API.Model
{
    /// <summary>
    /// Script execution Mode 
    /// </summary>
    public enum RunMode
    {
        /// <summary>
        /// performs the simulation moving the robot (default)
        /// </summary>
        Simulate = 1,

        /// <summary>
        /// Performs a quick check to validate the robot movements.
        /// </summary>
        QuickValidate = 2,

        /// <summary>
        /// Makes the robot program.
        /// </summary>
        MakeRobotProgram = 3,

        /// <summary>
        /// Makes the robot program and updates it to the robot.
        /// </summary>
        MakeRobotProgramAndUpload = 4,

        /// <summary>
        /// Makes the robot program and starts it on the robot (independently from the PC).
        /// </summary>
        MakeRobotProgramAndStart = 5,

        /// <summary>
        /// Moves the real robot from the PC (PC is the client, the robot behaves like a server).
        /// </summary>
        RunRobot = 6 
    }
}

File: Java/API/src/main/java/com/robodk/api/RunMode.java
package com.robodk.api;

/**
 * Run mode (behavior) of a RoboDK station: simulate movements, validate them quickly, generate a
 * robot program (offline programming), or run on the physical robot (online programming). See
 * {@link RoboDK#setRunMode(RunMode)} and {@link RoboDK#getRunMode()}.
 * <p>
 * The numeric values match the RoboDK API wire protocol and the reference RoboDK C# and Python
 * APIs, so they must not be changed.
 */
public enum RunMode {

    /** Simulates the robot movements (default). */
    SIMULATE(1),
    /** Performs a quick check to validate the robot movements. */
    QUICK_VALIDATE(2),
    /** Generates the robot program (offline programming). */
    MAKE_ROBOT_PROGRAM(3),
    /** Generates the robot program and uploads it to the robot controller. */
    MAKE_ROBOT_PROGRAM_AND_UPLOAD(4),
    /** Generates the robot program and starts it on the robot controller. */
    MAKE_ROBOT_PROGRAM_AND_START(5),
    /** Moves the physical robot from the PC (online programming). */
    RUN_ROBOT(6);

    private final int value;

    RunMode(int value) {
        this.value = value;
    }

    /** Returns the integer value used by the RoboDK API wire protocol for this run mode. */
    public int getValue() {
        return value;
    }

    /**
     * Resolves a {@link RunMode} from its RoboDK API wire protocol integer value.
     *
     * @param value the wire protocol value, as returned by RoboDK
     * @return the matching run mode, or {@link #SIMULATE} if the value is unknown
     */
    public static RunMode fromValue(int value) {
        for (RunMode runMode : values()) {
            if (runMode.value == value) {
                return runMode;
            }
        }
        return SIMULATE;
    }
}
